src/PactNet.Native/NativePactMessageBuilder.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections.Generic;
+
+using Newtonsoft.Json;
+
+using PactNet.Native.Exceptions;
+using PactNet.Native.Models;
+
+namespace PactNet.Native
+{
+    /// <summary>
+    /// Mock request builder
+    /// </summary>
+    public class NativePactMessageBuilder : IPactMessageBuilderV3
+    {
+        private readonly IMessageMockServer server;
+        private readonly MessagePactHandle pact;
+        private readonly PactConfig config;
+        private readonly MessageHandle message;
+        private readonly JsonSerializerSettings defaultSettings;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="NativePactMessageBuilder"/> class.
+        /// </summary>
+        /// <param name="server">Mock server</param>
+        /// <param name="pact">the pact handle</param>
+        /// <param name="config">the pact configuration</param>
+        /// <param name="defaultSettings">Default JSON serializer settings</param>
+        internal NativePactMessageBuilder(IMessageMockServer server, MessagePactHandle pact, PactConfig config, JsonSerializerSettings defaultSettings = null)
```

Comment:
The JSON settings should be removed because we should be using the settings defined on the `PactConfig`. The extension method that constructs this class doesn't actually pass them in, so these will always be `null` in real usage.